lib/Dialect/Transforms/IsolateGroupOpsPass.cpp

```diff
@@ -100,7 +112,8 @@ class TcpIsolateGroupOpsPass
     MLIRContext *context = op->getContext();
     RewritePatternSet patterns(context);
 
-    patterns.add<IsolateGroups>(context);
+    auto shouldCopyConstPredicate = [&](tcp::GroupOp, Value) { return true; };
+    populateIsolateGroupPatterns(patterns, shouldCopyConstPredicate);
```

Comment:
Just for my understanding, is the idea here that in the downstream (internal) use we'd define a separate pass that calls `populateIsolateGroupPatterns` with a custom `shouldCopyConstPredicate` functor? Was there an alternative that also avoids the duplication of the pass definition itself?

Reply:
Yes. that would be the idea. Looking around (admittedly not very hard), I did not find a way to add a predicate function as an argument to a pass constructor. Admittedly the downstream pass will be a bit bare...